store/src/main/java/com/nytimes/android/external/store/base/impl/RepeatWhenEmits.java

```diff
@@ -0,0 +1,41 @@
+package com.nytimes.android.external.store.base.impl;
+
+import javax.annotation.Nonnull;
+
+import rx.Observable;
+import rx.functions.Func1;
+
+import static com.nytimes.android.external.cache.Preconditions.checkNotNull;
+
+/**
+ * A Transformer that takes a source observable and re-subscribes to the upstream Observable when
+ * it emits.
+ */
+final class RepeatWhenEmits<T> implements Observable.Transformer<T, T> {
+
+    private final Observable source;
+
+    private RepeatWhenEmits(@Nonnull Observable source) {
+        this.source = source;
+    }
+
+    @Nonnull
+    static <T> RepeatWhenEmits<T> from(@Nonnull Observable source) {
+        return new RepeatWhenEmits<>(checkNotNull(source));
+    }
+
+    @Override
+    public Observable<T> call(Observable<T> upstream) {
```

Comment:
It was not your error, it was mine while renaming the signatures.

For understanding check [this post](http://blog.danlew.net/2016/01/25/rxjavas-repeatwhen-and-retrywhen-explained/) from Dan Lew.